src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -397,6 +420,23 @@ partition_downloader::download_log_with_capped_size(
     vlog(_ctxlog.info, "Starting log download with size limit at {}", max_size);
     gate_guard guard(_gate);
 
+    std::deque<segment_meta> staged_downloads;
+    model::offset start_offset{0};
+    model::offset_delta start_delta{0};
```

Comment:
If `max_size` is something like 0 bytes and we don't iterate through any downloads, these are left as 0, and the offset translator is initialized to -1, and the min/max offsets are left initialized to extrema.

Reply:
Looking at the shallow copy path, it seems we'll use the manifest offset just fine, but I'm not sure the offset translator state is correct